### ISSUE TITLE
WASGI Interface needed?

### ISSUE THREAD

**Issue by mikeAdamss** [GSS-Cogs/linked-data-repository#39]
Investigate production ready best practices with Sanic.

As I understand it, you don't _need_ a wsgi server (i.e gunicorn) in the way that other python frameworks tend to, but you **can** use one - so that leaves a few questions to answer:

* is that correct?
* should we be using one. anyway?
* if not, what are the "production" settings for the in-build sanic http server?

the spike is

- [ ] answer the above questions
- [ ] implement some reasonable variation of the now clear (via the above) "best practice"

**Comment by mikeAdamss** [comment]
@robons - this came up in standup - I'm genuinely not sure how accurate the thing I read was but a little time before we make any big choices seemed sensible.

**Comment by roblinksdata** [comment]
> @robons - this came up in standup - I'm genuinely not sure how accurate the thing I read was but a little time before we make any big choices seemed sensible.

I *suspect* that the way sanic works is it allows you to run a local dev environment (to simplify the dev process) - but I suspect that it's single threaded.

I'm very keen to ensure we're able to cope with multiple requests concurrently, so if the sanic web server itself can deal with that, then that's fine, else I think we'll need the ASGI interface (#40).

**Comment by mikeAdamss** [comment]
yeah that's typically the approach but stumbled over a few posts strongly of the opinion that Sanic is productionized in of itself (new to me, but so is Sanic). It won't be a long spike but worth doing
